File: backend/platform-contracts/src/main/java/univ/lille/dto/access/AccessResponseDTO.java
package univ.lille.dto.access;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class AccessResponseDTO {
    private boolean granted ; 
    private String reason ; 
    private String zoneName; 
        
}

File: mobile/lib/features/access/data/dto/access_response_dto.dart
class AccessResponseDTO {
  final bool granted;
  final String reason;
  final String zoneName;

  AccessResponseDTO({
    required this.granted,
    required this.reason,
    required this.zoneName,
  });

  factory AccessResponseDTO.fromJson(Map<String, dynamic> json) {
    return AccessResponseDTO(
      granted: json['granted'] as bool,
      reason: json['reason'] as String,
      zoneName: json['zoneName'] as String,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'granted': granted,
      'reason': reason,
      'zoneName': zoneName,
    };
  }
}